openpype/plugins/publish/integrate_version_attrs.py

```diff
@@ -0,0 +1,92 @@
+import pyblish.api
+import ayon_api
+
+from openpype import AYON_SERVER_ENABLED
+from openpype.client.operations import OperationsSession
+
+
+class IntegrateVersionAttributes(pyblish.api.ContextPlugin):
+    """Integrate version attributes from predefined key.
+
+    Any integration after 'IntegrateAsset' can fill 'versionAttributes' with
+    attribute key & value to be updated on created version.
+
+    The integration must make sure the attribute is available for the version
+    entity otherwise an error would be raised.
+
+    Example of 'versionAttributes':
+        {
+            "ftrack_id": "0123456789-101112-131415",
+            "syncsketch_id": "987654321-012345-678910"
+        }
+    """
+
+    label = "Integrate Version Attributes"
+    order = pyblish.api.IntegratorOrder + 0.5
+
+    def process(self, context):
+        available_attributes = ayon_api.get_attributes_for_type("version")
+        skipped_attributes = set()
+        project_name = context.data["projectName"]
+        op_session = OperationsSession()
+        for instance in context:
+            label = self.get_instance_label(instance)
+            version_entity = instance.data.get("versionEntity")
+            if not version_entity:
+                continue
+            attributes = instance.data.get("versionAttributes")
+            if not attributes:
+                self.log.debug((
+                    "Skipping instance {} because it does not have"
+                    " specified version attributes to set."
+                ).format(label))
```

Comment:
Just thought of this shorter version that might be nice:
```suggestion
                self.log.debug((
                    "Skipping instance {} because it does not"
                    " specify version attributes to set."
                ).format(label))
```